### PR TITLE
Support base/head branch

### LINKED ISSUE #8: Filter PRs by title string
Hi @juliangruber, thank you for creating this great actions!
I have a trouble with cases where there are multiple PRs. I would like to filter PRs by regular expressions, start_with, and so on for the title.

### PR DESCRIPTION
**Pull request by civitaspo**
Hi @juliangruber, I wrote some codes that supports base branch and head branch. This resolves #8 because a pull request is unique if the base branch and the head branch are defined, I thought this is better than filtering by title.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -5,15 +5,21 @@ const { GitHub, context } = require('@actions/github')
 
 const main = async () => {
   const token = core.getInput('github-token')
-  const branch = core.getInput('branch')
+  const head_branch = core.getInput('branch', { required: false }) || core.getInput('head_branch')
```
Comment (civitaspo) [inline index.js:8]: This line follows the backward compatibility for `branch` and `head_branch`.

File: index.js [hunk 2]
```diff
@@ -5,15 +5,21 @@ const { GitHub, context } = require('@actions/github')
...
+  const base_branch = core.getInput('base_branch', { required: false })
 
   const octokit = new GitHub(token)
 
-  const res = await octokit.pulls.list({
+  let params = {
     ...context.repo,
     state: 'open',
     head: `${context.repo.owner}:${branch}`
-  })
```
Comment (juliangruber) [inline index.js:16]: can be refactored to

```js
const res = await octokit.pulls.list({
  ...context.repo,
  state: 'open',
  head: `${context.repo.owner}:${branch}`,
  base
})
```

File: index.js [hunk 3]
```diff
@@ -5,15 +5,21 @@ const { GitHub, context } = require('@actions/github')
 
 const main = async () => {
   const token = core.getInput('github-token')
-  const branch = core.getInput('branch')
+  const head_branch = core.getInput('branch', { required: false }) || core.getInput('head_branch')
+  const base_branch = core.getInput('base_branch', { required: false })
```
Comment (juliangruber) [inline index.js:9]:
```suggestion
  const base = core.getInput('base', { required: false })
```

File: index.js (hunk 1 above)
Comment (juliangruber) [inline index.js:8]:
```suggestion
  const branch = core.getInput('branch')
```

The argument pair `branch` and `base` seem fine too

Comment (juliangruber) [reply index.js:8]: Also, in your PR, the variable `branch` is used but not defined. Accepting the suggested changes should fix that

**Review by juliangruber — CHANGES_REQUESTED** [review]

**Comment by civitaspo** [comment]
@juliangruber Thanks for your rapid review! I applied your reviews, so please review again.

**Review by juliangruber — CHANGES_REQUESTED** [review]
Looks great! Could you please also add the `base` input to https://github.com/juliangruber/find-pull-request-action/blob/master/action.yml?

**Comment by civitaspo** [comment]
@juliangruber Thanks, I added the option to `action.yml`.

**Comment by juliangruber** [comment]
Thank you :) https://github.com/juliangruber/find-pull-request-action/releases/tag/v1.2.0

**Pull request merged**